pyresttest/validators.py

```diff
@@ -289,6 +290,24 @@ def parse(cls, config, extractor_base=None):
         return cls.configure_base(config, base)
 
 
+class CookieExtractor(AbstractExtractor):
+    """ Extractor that pulls out a named cookie """
+    extractor_type = 'cookie'
+    is_header_extractor = True
+
+    def extract_internal(self, query = None, args = None, body = None, headers = None):
+        extracted = [y[1] for y in filter(lambda x: x[0] == 'set-cookie', headers) if y[1].startswith(query + '=')]
+        if len(extracted) == 0:
+            raise ValueError("Cookie named {0} not known".format(query))
+        else:
+            return Cookie.SimpleCookie(extracted[-1])[query].value
```

Comment:
I'm not as familiar with the syntax of the cookie library, but this *seems* solid.  Would have to test it a bit though.